ClosedXML_Tests/Excel/Worksheets/XLWorksheetTests.cs

```diff
@@ -234,5 +234,83 @@ public void CanCopySheetsWithAllAnchorTypes()
                 ws3.CopyTo("Copy4");
             }
         }
+
+        [Test]
+        public void WorksheetNameCannotStartWithApostrophe()
+        {
+            var title = "'StartsWithApostrophe";
+            TestDelegate addWorksheet = () =>
+            {
+                using (var wb = new XLWorkbook())
+                {
+                    wb.Worksheets.Add(title);
+                }
+            };
+
+            Assert.Throws(typeof(ArgumentException), addWorksheet);
+        }
+
+        [Test]
+        public void WorksheetNameCannotEndWithApostrophe()
+        {
+            var title = "EndsWithApostrophe'";
+            TestDelegate addWorksheet = () =>
+            {
+                using (var wb = new XLWorkbook())
+                {
+                    wb.Worksheets.Add(title);
+                }
+            };
+
+            Assert.Throws(typeof(ArgumentException), addWorksheet);
+        }
+
+        [Test]
+        public void WorksheetNameCanContainApostrophe()
+        {
+            var title = "With'Apostrophe";
+            var savedTitle = "";
+            TestDelegate saveAndOpenWorkbook = () =>
+            {
+                using (var ms = new MemoryStream())
+                {
+                    using (var wb = new XLWorkbook())
+                    {
+                        wb.Worksheets.Add(title);
+                        wb.Worksheets.First().Cell(1, 1).FormulaA1 = $"{title}!A2";
+                        wb.SaveAs(ms);
+                    }
+
+                    using (var wb = new XLWorkbook(ms))
+                    {
+                        savedTitle = wb.Worksheets.First().Name;
+                    }
+                }
+            };
+
+            Assert.DoesNotThrow(saveAndOpenWorkbook);
+            Assert.AreEqual(title, savedTitle);
+        }
+
+        /// <summary>
+        /// Excel escapes symbol ' in worksheet title so we have to process this correctly.
+        /// </summary>
+        [Test]
+        public void CanOpenWorksheetWithEscapedApostrophe()
```

Comment:
I think this test belongs with the others in `.\ClosedXML_Tests\Excel\Loading\LoadingTests.cs`. Do you agree?

Reply:
Yes, I agree. Will move it.